### ISSUE TITLE
Displayname / Full Name not displayed / searchable

### ISSUE THREAD

**Issue by linuxrrze** [nextcloud/user_saml#119]
### Steps to reproduce
1. Install nextcloud
2. Configure user_saml auth
2. Set saml-attribute-mapping-displayName_mapping / saml-attribute-mapping-email_mapping

### Expected behaviour
When opening user "Personal" view ( /index.php/settings/personal)
"Full name" should show the users "displayName" attribute value

### Actual behaviour
Only the Login(name) is displayed - no "displayName".

The attribute mapping seems to work fine with the email attribute:
This is displayed under "Email" as expected.

Another thing to mention is, that when logged in as Administrator, and
viewing the user list (/settings/users) the "Full name" attribute is displayed
correctly - so the required data seems to be correctly transferred.

What makes things worse is, that this also prevents searching for people,
unless you know their login name.

The user_ldap backend uses something like 
  "Displayname" (usually "Firstname Lastname")
or - even better (but optional)
  "Displayname (Loginname)"

It'd be great to have something like this in user_saml.

### Server configuration
Nextcloud 12.0.0

**Operating system**:
Ubuntu 16.04

**Web server:**
Apache

**Database:**
postgresql 9.5

**PHP version:**
7.0

**Nextcloud version:** (see Nextcloud admin page)
12.0.0 (was the same in 11.0.x)

**Where did you install Nextcloud from:**
Web

**List of activated apps:**

```
Enabled:
  - activity: 2.5.2
  - bruteforcesettings: 1.0.2
  - comments: 1.2.0
  - dav: 1.3.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - onlyoffice: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - updatenotification: 1.2.0
  - user_saml: 1.3.2
  - workflowengine: 1.2.0
```

**Nextcloud configuration:**

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "www.mydomain.de"
        ],
        "datadirectory": "\/proj.stand\/web\/apps\/nextcloud\/instances\/saml\/nextcloud-latest\/data",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbtype": "sqlite3",
        "version": "12.0.0.29",
        "logtimezone": "UTC",
        "installed": true,
        "instanceid": "oc8ip83453459ru",
        "loglevel": 2,
        "maintenance": false
    }
}
```

**user_saml configuration:**

```
{
    "apps": {
        "user_saml": {
            "enabled": "yes",
            "installed_version": "1.3.2",
            "types": "authentication",
            "type": "saml",
            "idp-entityId": "https:\/\/www.sso.mydomain.de\/simplesaml\/saml2\/idp\/metadata.php",
            "idp-singleSignOnService.url": "https:\/\/www.sso.mydomain.de\/simplesaml\/saml2\/idp\/SSOService.php",
            "idp-singleLogoutService.url": "https:\/\/www.sso.mydomain.de\/simplesaml\/saml2\/idp\/SingleLogoutService.php",
            "general-uid_mapping": "urn:mace:dir:attribute-def:uid",
            "general-require_provisioned_account": "0",
            "idp-x509cert": "\n-----BEGIN CERTIFICATE-----REMOVED\n-----END CERTIFICATE-----",
            "saml-attribute-mapping-displayName_mapping": "urn:mace:dir:attribute-def:displayName",
            "saml-attribute-mapping-email_mapping": "urn:mace:dir:attribute-def:mail"
        }
    }
}

```

**Comment by linuxrrze** [comment]
Just found closely related issue #125

**Comment by linuxrrze** [comment]
Any progress on this?

I was looking into the `user_ldap` code, hoping to find the code to fix this in `user_saml`.
I hoped the `user_*` modules would look quite similar, but had to realize that's not the case.

So it'd be very helpful if someone could give a hint where to look/what to change in order to fix this issue.

**Comment by LukasReschke** [comment]
https://github.com/nextcloud/user_saml/pull/137 should probably resolve this.

**Comment by LukasReschke** [comment]
Testing appreciated 😄

**Comment by linuxrrze** [comment]
I just tried latest git `user_saml` with `nextcloud 12.0.0`:

"Full Name" in Personal view still shows the wrong value (not `DisplayName`)

**Comment by linuxrrze** [comment]
Test repeated with nextcloud 12.01RC4:

"Full Name" in "Personal" view still shows the wrong value (not DisplayName)

Data is obviously delivered with both nextcloud versions as the correct "Full name" is listed when viewing "Users" as administrator.

Unless someone can confirm, that these patches actually work I'd suggest to re-open this issue.

**Comment by linuxrrze** [comment]
Ok - maybe the problem is not `user_saml` app specific, but a general nextcloud server issue:

I just replaced one line in `settings/personal.php`, and I got the correct `displayName` in `Personal`:

```
+ $tmpl->assign('displayName', OC_User::getDisplayName(OC_User::getUser()));
- $tmpl->assign('displayName', $userData[\OC\Accounts\AccountManager::PROPERTY_DISPLAYNAME]['value']);
```

I don't know if this is the correct way to do it (even the called function name was just a wild guess).
This may also break other `user_*` backends, but I hope someone with more knowledge about nextcloud internals can comment on this.

BTW - does anyone read comments on closed issues?
